app/actions/remote/groups.ts

```diff
@@ -4,13 +4,32 @@
 import {Client} from '@client/rest';
 import NetworkManager from '@managers/network_manager';
 
-export const getGroupsForAutocomplete = async (serverUrl: string, channelId: string) => {
+export const fetchGroupsForChannel = async (serverUrl: string, channelId: string) => {
     let client: Client;
     try {
         client = NetworkManager.getClient(serverUrl);
+        return client.getAllGroupsAssociatedToChannel(channelId);
     } catch (error) {
-        return [];
+        return undefined;
```

Comment:
Why ubdefined and not {error}, same for the others.. including the one returning an empty array

Reply:
Mostly to make the methods simpler where the fetch method is being called; the `fetch...` actions return either `Promise<Model>` or `undefined`, so i'm able to do things like `const data = await fetchSomeThing(); data?.myThing.map(...`.

For `autoComplete`, it plugs back into an array of sections to display the `@` mentions; it's much simpler to pass back an empty `[]` - we don't necessarily mind too much in this specific place that groups fetching may have failed.

There are three other chunks of work (group data handling, and then group push notification handling, post mention highlighting) where I'll be expanding the `catch(e){ ... }` to check what the error is, and retry, and potentially check the local db, or log the error and exit without asking the calling method to handle the error.

Reply:
For groups - we can pass the `{error}` to the method calling it, but there isn't much we  plan on doing with the `error` anyway - unless we specifically want to show to the user in the `@` mention list that the autocomplete for groups may have failed.

Reply:
Discussed with @enahum - will return `{error}` to stay consistent with other actions, and handle on the calling method 👍